File: fm-dddrive/src/main/java/io/dddrive/core/property/model/EnumSetProperty.java
package io.dddrive.core.property.model;

import java.util.Set;

import io.dddrive.core.enums.model.Enumerated;

public interface EnumSetProperty<E extends Enumerated> extends Property<E> {

	Set<E> getItems();

	boolean hasItem(E item);

	void clearItems();

	void addItem(E item);

	void removeItem(E item);

}

File: fm-dddrive/src/main/java/io/dddrive/core/property/model/EnumSetProperty.kt
package io.dddrive.core.property.model

import io.dddrive.core.enums.model.Enumerated

interface EnumSetProperty<E : Enumerated> : Property<E> {

	val items: Set<E>

	fun hasItem(item: E): Boolean

	fun clearItems()

	fun addItem(item: E)

	fun removeItem(item: E)

}
